File: BusRider/Model/JWSearchRequest.m
//
//  JWSearchRequest.m
//  BusRider
//
//  Created by John Wong on 12/17/14.
//  Copyright (c) 2014 John Wong. All rights reserved.
//

#import "JWSearchRequest.h"

@implementation JWSearchRequest

- (NSDictionary *)params {
    return @{
             @"LsName": self.keyWord ? : @""
             };
}

- (NSString *)actionName {
    return @"query!search";
}

- (NSString *)validateParams {
    if (!self.keyWord || self.keyWord.length <= 0) {
        return @"搜索关键字不能为空";
    }
    return nil;
}

@end

File: BusRider WatchKit Extension/JWSearchRequest.swift
//
//  JWSearchRequest.swift
//  BusRider
//
//  Created by John Wong on 5/2/15.
//  Copyright (c) 2015 John Wong. All rights reserved.
//

import UIKit

class JWSearchRequest: JWRequest {
   
    var keyword = ""
    
    override func params() -> Dictionary<NSObject, AnyObject> {
        let encodedKeyword: String = self.keyword.stringByAddingPercentEncodingWithAllowedCharacters(NSCharacterSet())!
        return [
            "LsName": encodedKeyword
        ]
    }
    
    override func actionName() -> String {
        return "query!search"
    }
}
